pkg/utils/v1alpha1/kubernetes.go

```diff
@@ -101,8 +102,9 @@ func GetCSIVolume(volumeID string) (*apis.CSIVolume, error) {
 	if err != nil {
 		return nil, err
 	}
+
 	if len(volList.Items) == 0 {
-		return nil, nil
+		return nil, fmt.Errorf("orphaned iscsi session exists for volume %s", volumeID)
```

Comment:
Let's return IsNotFound error over here.

Reply:
IsNotFound returns boolean , we can simply add `volume not found %s ,volumeID` error